contributors/design-proposals/storage/local-storage-pv.md

```diff
@@ -761,3 +761,138 @@ The central manager will poll each worker for their discovered volumes and creat
 them.  When a PV is released, then it will send the cleanup request to the worker.
 
 Detailed design TBD
+
+
+### Phase 3: 1.11 Alpha
+
+#### Dynamic Provisioning
+
+The local volume provisioner will be enhanced to support dynamic provisioning of local storage,
+which depends on the newly introduced changes in [Volume Topology-aware Scheduling](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/storage/volume-topology-scheduling.md),
+that is:
+
+1. When the feature gate `DynamicProvisioningScheduling` is enabled,
+the scheduler will verify a pod's volume requirements,
+and set an annotation `annScheduledNode` on PVCs that need provisioning.
+2. New fields `Capacity` will be added in `Status` of `StorageClass` to
+allow storage providers to report capacity limits per topology domain.
+
+The local volume provisioner needs to be updated in several areas to support dynamic provisioning.
+
+##### Backend storage management
+
+Mechanisms to manage local storage sources are varied. They could be LVM, Ceph, or something like that.
+In addition to current `populator`, `deletor` and `discoverer`, a new component named
+`storageManager` will be introduced in the plugin, to manage the backend storage.
+
+`storageManager` is a general interface, each kind of storage should have its own implementation.
+During start-up, a new environment variable `STORAGE_BACKEND` is set in the provisioner DaemonSet,
+to determine which StorageManager instance to start.
+(TBD: Should we support the scenario of multi backends on a node?)
+
+A StorageManager instance is responsible for:
+
+1. Monit changes in storage input, and handle corresponding initialization.
+2. Update capacity in StorageClass when storage source changed.
+3. Provide the ability to create volumes, detailed in the volume creation section.
+
+Take LVM as an example, when a new source disk "/dev/new" is added to its StorageClass `exStorage`,
+the StorageManager should:
+
+1. Check if related PV (physical volume in LVM) exists, if not, create one with command:
+```
+pvcreate /dev/new
+```
+
+2. Check if volume group exStorageGroup exists, if not, create it:
+```
+vgcreate exStorageGroup /dev/new
+```
+
+
+3. If VG exists, check if the physical volume exists in volume group exStorageGroup,
+if not, add it:
+```
+vgextend exStorageGroup /dev/new
+```
+
+4. Update capacity in StorageClass if there is a change in VG size.
+
+A new field `storageSource` will be added into the configuration ConfigMap, to illustrate available
+storage source for provisioning.
+
+```
+kind: ConfigMap
+metadata:
+  name: local-volume-config
+  namespace: kube-system
+data:
+ storageClassMap: |
+    local-fast:
+      hostDir: "/mnt/ssds"
+    local-slow:
+      hostDir: "/mnt/hdds"
+    local_dynamic:
+      storageSources:

```

Comment:
I'd say only the name of a VG should be specified for things to work. Nothing more required.